homeassistant/config.py

```diff
@@ -859,7 +870,9 @@ async def async_process_component_config(  # noqa: C901
                 await config_validator.async_validate_config(hass, config)
             )
         except (vol.Invalid, HomeAssistantError) as ex:
-            async_log_exception(ex, domain, config, hass, integration.documentation)
+            async_log_exception(
+                ex, domain, config, hass, integration.documentation, raise_on_failure
+            )
             return None
```

Comment:
This is weird, why would we log if the caller asked us to raise?
This seems more reasonable:
```suggestion
            if raise_on_failure:
                raise
            async_log_exception(ex, domain, config, hass, integration.documentation)
            return None
```